src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -931,8 +931,17 @@ winrt::fire_and_forget AppHost::_peasantNotifyActivateWindow()
     const auto peasant = _peasant;
     const auto hwnd = _window->GetHandle();
 
+    auto weakThis{ weak_from_this() };
+
     co_await winrt::resume_background();

```

Comment:
Does this need to run on the BG thread in the first place?

Reply:
I had it on a background thread so that a hung Monarch wouldn't hang a peasant process when the peasant gets activated. 

I suppose these days, it's all the same process, so we could pull this one.